tiledb/sm/group/group_directory.cc

```diff
@@ -0,0 +1,365 @@
+/**
+ * @file   group_directory.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class GroupDirectory.
+ */
+
+#include "tiledb/sm/group/group_directory.h"
+#include "tiledb/common/logger.h"
+#include "tiledb/common/stdx_string.h"
+#include "tiledb/sm/filesystem/vfs.h"
+#include "tiledb/sm/misc/parallel_functions.h"
+#include "tiledb/sm/misc/utils.h"
+#include "tiledb/sm/misc/uuid.h"
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+/* ********************************* */
+/*     CONSTRUCTORS & DESTRUCTORS    */
+/* ********************************* */
+
+GroupDirectory::GroupDirectory(
+    VFS* vfs,
```

Comment:
Why do we take a VFS and threadpool here, while the other ones get a storage manager?

Reply:
This is treated the same as ArrayDirectory. We don't need the full storage manager so taking the references to the components was seen as better limiting the scope.